app/components/course_details/view.rb

```diff
@@ -60,6 +61,10 @@ def require_course_type?
       !trainee.draft?
     end
 
+    def require_study_mode?
```

Comment:
I think we can just `delegate :require_study_mode?, to: :trainee` and remove the method definition here